d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperAnnouncer.java

```diff
@@ -53,6 +55,9 @@
 
 public class ZooKeeperAnnouncer
```

Comment:
1. Can we add some unit tests in `ZookeeperConnectionManagerTest` for warm-up scenarios?
2. Have you tested the flow locally? You can publish a SNAPSHOT version of pegasus and try with local ZooKeeper. Please let me know if you need any help onthis

Reply:
Thank you for reviewing.
1. Added unit tests in ZookeeperConnectionManagerTest for warm-up scenarios. 
2. Have tested the flow locally using a SNAPSHOT version of pegasus, with a SNAPSHOT version of container for testing dark warm-up feature with local ZooKeeper for a midtier service.